tests/monotouch-test/System.Net.Http/NetworkResources.cs

```diff
@@ -42,23 +49,64 @@ public static class NetworkResources {
 			VisualStudioRobotsUrl,
 		};
 
+		static Dictionary<string, bool> sites = new Dictionary<string, bool> ();
+		static bool VerifyNetworkConnection (string url, int timeoutInSeconds = 5)
+		{
+			bool value;
+			lock (sites) {
+				if (sites.TryGetValue (url, out value)) {
+					return value;
+				}
+			}
+			value = VerifyNetworkConnectionImpl (url, timeoutInSeconds);
+			lock (sites) {
+				sites [url] = value;
+			}
+			return value;
+		}
+
+		static bool VerifyNetworkConnectionImpl (string url, int timeoutInSeconds = 5)
+		{
+			try {
+				var client = new HttpClient ();
+				var task = client.GetStringAsync (url);
+				if (task.Wait (TimeSpan.FromSeconds (timeoutInSeconds)))
+					return task.IsCompletedSuccessfully;
+				TestRuntime.NSLog ($"VerifyNetworkConnection ({url}) => task timed out, status: {task.Status}");
+				return false;
+			} catch (Exception e) {
+				TestRuntime.NSLog ($"VerifyNetworkConnection ({url}) => {e}");
+				return false;
+			}
+		}
+
+		static string AssertNetworkConnection (string url)
+		{
+			if (!VerifyNetworkConnection (url)) {
+				TestRuntime.NSLog ($"AssertNetworkConnection ({url}): failure");
+				Assert.Ignore ($"The site {url} is not accessible. This test will be ignored.");
+			}
+			return url;
+		}
+
 		public static class Httpbin {
-			public static readonly string Url = "https://httpbin.org";
-			public static readonly Uri Uri = new Uri ("https://httpbin.org");
-			public static readonly string DeleteUrl = "https://httpbin.org/delete";
-			public static readonly string GetUrl = "https://httpbin.org/get";
-			public static readonly string PatchUrl = "https://httpbin.org/patch";
-			public static readonly string PostUrl = "https://httpbin.org/post";
-			public static readonly string PutUrl = "https://httpbin.org/put";
-			public static readonly string CookiesUrl = $"https://httpbin.org/cookies";
-			public static readonly string HttpUrl = "http://httpbin.org";
+			static bool? isHttpbinDown;
 
+			public static string Url { get { return AssertNetworkConnection ("https://httpbin.org"); } }
```

Comment:
Nit but why not use a lambda expression here as well? Looks like it was previously a readonly var so unless there is potential for the Url property to be set later it can be a lambda?
```suggestion
			public static string Url => AssertNetworkConnection ("https://httpbin.org");
```

Reply:
The only reason is that I started using C# before lambda expressions were born, so I don't usually think to use them :)